nodebuilder/p2p/flags.go

```diff
@@ -68,12 +69,18 @@ func ParseFlags(
 // and returns either the parsed network or the build's default network
 func ParseNetwork(cmd *cobra.Command) (Network, error) {
 	parsed := cmd.Flag(networkFlag).Value.String()
+	if parsed == DefaultNetwork.String() {
+		return DefaultNetwork, nil
+	}
 	// no network set through the flags, so check if there is an override in the env
 	if parsed == "" {
 		envNetwork, err := parseNetworkFromEnv()
 		// no network found in env, so use the default network
 		if envNetwork == "" {
-			return DefaultNetwork, err
+			if err != nil {
+				return "", err
+			}
+			return "", fmt.Errorf("no network provided, allowed values: %s", listProvidedNetworks())
 		}
 		return envNetwork, err
 	}
```

Comment:
I just realized this expects an empty flag and only triggers the Env Var parsing. With this change, env var will be ignored, which is a bug. For this fix to work, I think we need to prioritize Env Var, and only if it's not defined to look into flags, which needs a slight change in the logic here. We will also need to add t the flag description that the flag is overridden by the env var

I think this is what I meant in a previous comm 
> We need to be careful, though, not to break behaviors

Reply:
I think right priority could be 

- Command line arguments
- Environment variables
- Defaults

With this, we would fallback onto defaults; and the args would always override the env vars. WDYT ?

Reply:
I think you are right about this priority. The tricky part is not to treat the default coming from the flag as the command line argument overriding the env var. But that's doable

Reply:
If we make `if flag == default => return env_var.network`, then there will be a catch, and the following example won't follow the priority, as the flag won't override the env var.
`CELESTIA_CUSTOM=arabica celestia light start --p2p.network celestia`

Reply:
@Wondertan I have re-structured the network parsing .. And it make more sense to override with custom network. Lets look at some of the examples:

> celestia full|light start|init
`celestia`

> celestia full|light start|init --p2p.network=""
`error`

> celestia full|light start|init --p2p.network="mocha"
`mocha`

> CELESTIA_CUSTOM="celestia" celestia full|light start|init --p2p.network="mocha"
`celestia`

> CELESTIA_CUSTOM="" celestia full|light start|init --p2p.network=""
`error`

Tests should cover these cases.

Reply:
In general, flags should be prioritised over end vars, but I agree that in our case it's fine to make envs a priority, as long as we document that properly on the command.